js/apps/admin-ui/src/realm-settings/OpenIdEndpointConigurationTab.tsx

```diff
@@ -0,0 +1,175 @@
+
+  const setupForm = () => {
+    let attributes = realm.attributes || {};
+    
+    if (!("claimsSupported" in attributes)) {
+      getOpenIdEndpointConfiguration().then((result) => {
+        attributes.claimsSupported = result.claims_supported.join(",");
+        setValue("attributes", attributes);
+      });
+    }
+    else{
+      setValue("attributes", attributes);
+    }
+    
+  };
+
+  useEffect(setupForm, []);
+
+  async function getOpenIdEndpointConfiguration(): Promise<any> {
+    const response = await fetch(
+      `${addTrailingSlash(adminClient.baseUrl)}/realms/${
+        realm.realm
+      }/.well-known/openid-configuration`,
+      {
+        method: "GET",
+        headers: getAuthorizationHeaders(await adminClient.getAccessToken()),
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error(
+        `Server responded with invalid status: ${response.statusText}`,
+      );
+    }
+    return response.json();
+  }
+
+  return (
+    <PageSection variant="light">
+      <FormAccess
+        isHorizontal
+        role="manage-realm"
+        className="pf-u-mt-lg"
+        onSubmit={handleSubmit(save)}
+      >
+        <FormGroup
+          label={t("claimsSupported")}
+          fieldId="kc-claims-supported"
+          labelIcon={
+            <HelpItem
+              helpText={t("claimsSupportedHelp")}
+              fieldLabelId="claimsSupported"
+            />
+          }
+        >
+          <Controller
+            name="attributes"
+            control={control}
+            render={({ field }) => {
+              const claimsSupported = field.value?.claimsSupported?.trim()
+                ? field.value.claimsSupported.split(",")
+                : [];
+              return (
+                <>
+                  {claimsSupported.map((claim: string, index: number) => {
+                    if (claim) {
+                      return (
+                        <InputGroup key={index}>
+                          <TextInput
+                            id={"textInput" + index}
+                            value={claim}
+                            onChange={(value) => {
+                              claimsSupported[index] = value;
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
+                              field.onChange(field.value);
+                            }}
+                            aria-label="input with button"
+                          />
+                          <Button
+                            id={"inputButtonRemove" + index}
+                            icon={<MinusIcon />}
+                            onClick={() => {
+                              claimsSupported.splice(index, 1);
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
```

Comment:
These will also mutate the original value in-place. This should be set using the `onChange` handler with a new value instead.

Reply:
Used 
```
let attributes = field.value || {};
claimsSupported.splice(index, 1);
attributes.claimsSupported = claimsSupported.join(",");
field.onChange(attributes);
```